src/zcash/Proof.hpp

```diff
@@ -258,20 +259,65 @@ class ProofVerifier {
     // verification, used when avoiding duplicate effort
     // such as during reindexing.
     static ProofVerifier Disabled();
-
+    
+    // Creates a batch verifier 
+    static ProofVerifier Batch();
+    static ProofVerifier RegularAndBatch();
+    //for a batch verifier the check method only adds proof to the batch and always returns true
     template <typename VerificationKey,
               typename ProcessedVerificationKey,
+              typename ProcessedBatchVerificationKey,
               typename PrimaryInput,
               typename Proof
               >
     bool check(
         const VerificationKey& vk,
         const ProcessedVerificationKey& pvk,
+        const ProcessedBatchVerificationKey& pbvk,
         const PrimaryInput& pi,
         const Proof& p
     );
+    //temporary method checks batch without adding more proofs to batch
+    template <typename ProcessedBatchVerificationKey>
+    bool checkBatch(const ProcessedBatchVerificationKey& pbvk);
 };
 
+/*class BatchVerifier: public ProofVerifier {
```

Comment:
Were there problems implementing `BatchVerifier` as a subclass of `ProofVerifier`?